modules/renter/persist.go

```diff
@@ -234,17 +244,19 @@ func (r *Renter) load() error {
 		return err
 	}
 
-	// Load contracts, repair set, and entropy.
-	data := struct {
-		Tracking  map[string]trackedFile
-		Repairing map[string]string // COMPATv0.4.8
-	}{}
-	err = persist.LoadJSON(saveMetadata, &data, filepath.Join(r.persistDir, PersistFilename))
-	if err != nil {
-		return err
+	// Load contracts and entropy.
+	r.persist = persistence{
+		Tracking: make(map[string]trackedFile),
 	}
-	if data.Tracking != nil {
-		r.tracking = data.Tracking
+	err = persist.LoadJSON(settingsMetadata, &r.persist, filepath.Join(r.persistDir, PersistFilename))
+	if err != nil {
```

Comment:
`if err == persist.ErrBadVersion` I believe is fine here. Then you'll have to have an `else if err !- nil` after the statement to handle the rest of the cases. That way corrupt files are handled correctly right away instead of trying a bunch of recovery steps.